.github/workflows/matrix-deploy.yml

```diff
@@ -11,26 +11,68 @@ on:
 jobs:
 
   get-branch:
-
     runs-on: ubuntu-22.04
     name: Get Target Branch Name
     outputs:
       branch: ${{ steps.var.outputs.branch }}
-    
     steps:
       - id: var
         run: |
           GIT_BRANCH=${GITHUB_BASE_REF:-${GITHUB_REF#refs/heads/}}
           echo "branch=$GIT_BRANCH" >> $GITHUB_OUTPUT
 
   create-and-publish-image:
-
     name: Create & Publish Image
     uses: ./.github/workflows/build-image.yml
     secrets: inherit
 
+  check-for-terraform-changes:
+    needs: create-and-publish-image
+    runs-on: ubuntu-latest
```

Comment:
For all of these, should we be specifying `ubuntu-22.04` instead of `ubuntu-latest`? Just thinking it could reduce the risk where GitHub update the distro and it could have updated packages that have breaking changes for our runners (albeit unlikely)